python/cudf/cudf/core/dataframe.py

```diff
@@ -820,11 +825,39 @@ def fallback(col, fn):
                 else:
                     result[col] = fallback(rhs[col], _reverse_op(fn))
         elif isinstance(other, Series):
-            raise NotImplementedError(
-                "Series to DataFrame arithmetic not supported "
-                "until strings can be used as indices. Try converting your"
-                " Series into a DataFrame first."
+            other_cols = other.to_pandas().to_dict()
+            other_cols_keys = list(other_cols.keys())
+            result_cols = list(self.columns)
+            df_cols = list(result_cols)
+            for new_col in other_cols.keys():
+                if new_col not in result_cols:
+                    result_cols.append(new_col)
+            for col in result_cols:
+                if col in df_cols and col in other_cols_keys:
+                    l_opr = self._cols[col]
+                    r_opr = other_cols[col]
+                else:
+                    if col not in df_cols:
+                        r_opr = other_cols[col]
+                        l_opr = Series(
+                            column_empty(
+                                len(self),
+                                masked=True,
+                                dtype=np.array(r_opr).dtype,
```

Comment:
I believe we have a utility to get the dtype of a scalar. Numpy shouldn't be trusted here. I.e. say `r_opr` is 1, should it be `int8`, `int16`, `int32`, or `int64`?

Reply:
@brandon-b-miller this needs to be addressed before we can merge. We can't call `np.array` here. Given this stems from `other` which is a `cudf.Series`, could we just use `other.dtype`? If we're trying to handle `pandas.Series` objects with `object` dtype then things become trickier and we need to intelligently infer the dtype somehow.

Reply:
roger that, will update this.

Reply:
I think this might have actually been `other.dtype` at some point and accidentally scooped up in some commit.